index.js

```diff
@@ -124,8 +124,10 @@ function _readFC16(data, next) {
  * @private
  */
 function _writeBufferToPort(buffer) {
+    var transaction = this._transactions[this._transactionId]
+
     this._port.write(buffer);
-    this._timeoutHandle = _startTimeout(this._timeout, this._next);
+    this._timeoutHandle = _startTimeout(this._timeout, transaction.next);
```

Comment:
Each request needs to have its own timeout.

**Example**
Think of a timeout of 5 seconds.
* 0:00.00: send request to device 1 (transaction id TID = 1)
* 0:00.03: send request to device 2 (transaction id TID = 2)
  * overwrite timeout handle
* 0:00.04: receive response of TID 1
  * cancel timeout handle (which is the timeout handle of TID 2)
* 0:00.05: receive response of TID 2
  * no timeout handle to cancel
* 0:00.005: emit timeout for TID 1